### ISSUE TITLE
508-defect-1 [COGNITION, KEYBOARD]: Form wizard collapses unexpectedly

### ISSUE THREAD

**Issue by Mottie** [department-of-veterans-affairs/va.gov-team#19845]
# [508-defect-1](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## User Story or Problem Statement

As a keyboard user, I would expect the form elements within the wizard to not cause the wizard to collapse when I press <kbd>Enter</kbd> 

## Details

The wizard "Let's get started" button is acting like a submit button and being toggled when any form element "see" the <kbd>Enter</kbd> key and attempts to submit the form.

- The wizard collapsing is jarring & unexpected
- The expanded wizard should be persistent
- Focus is lost
- This problem occurs **in production** 

## Acceptance Criteria

- [ ] Prevent closing the wizard when <kbd>Enter</kbd> is pressed
- [ ] Maintain focus on the form element

## Environment

- Operating System: any
- Browser: any
- Screenreading device: any
- Server destination: production

## Steps to Recreate

1. Go to https://www.va.gov/disability/how-to-file-claim/
1. Click on the green "Let's get started" wizard button - it will turn blue and show the first wizard question.
1. Once open use the <kbd>Tab</kbd> to navigate to the active duty radio group
1. Press <kbd>Space</kbd> to make a choice and see the wizard reveal the next step.
1. While still focused on the radio, press <kbd>Enter</kbd>
1. The "Let's get started button" will once again turn green and collapse the wizard
1. Focus is lost, but will focus on the link after the wizard start button if <kbd>Tab</kbd> is pressed.
1. Clicking on the "Let's get started" button will reveal the wizard in its previous state.

## Proposed Solution (if known)

This problem may be fixed by adding a `type="button"` to the wizard toggle

```diff
    <button
+    type="button"
      aria-expanded={this.state.expanded ? 'true' : 'false'}
      aria-controls="wizardOptions"
      className={buttonClasses}
      onClick={() => this.setState({ expanded: !this.state.expanded })}
    >
      {buttonText}
    </button>
```
-->

## Screenshots or Trace Logs

https://user-images.githubusercontent.com/136959/107684824-dd4ad680-6c68-11eb-88fc-8652f44faed6.mov

**Comment by Mottie** [comment]
@joshkimux Fix has been merged, please verify and feel free to close this issue. Thanks!

**Comment by joshkimux** [comment]
@Mottie Dumb question-- was this merged on production? The behavior hasn't changed there unfortunately D:

**Comment by Mottie** [comment]
It was merged into staging yesterday.. we only do production updates on weekdays at 3pm ET. So you should see this fix in place later today.

**Comment by joshkimux** [comment]
@Mottie Awesome! I can't access it on staging atm since staging will send me to a separate wizard page, but I'll give it a check later this afternoon on prod! :D

**Comment by taylorkaren** [comment]
Removing the forms label as this is not a Forms Library issue.
